File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/dialogue/model/PersonaId.java
package com.study.webflux.rag.domain.dialogue.model;

public record PersonaId(
	String value
) {
	public static final PersonaId DEFAULT = new PersonaId("default");

	public PersonaId {
		if (value == null || value.isBlank()) {
			throw new IllegalArgumentException("personaId cannot be null or blank");
		}
		if (value.length() > 64) {
			throw new IllegalArgumentException("personaId too long");
		}
	}

	public static PersonaId of(String value) {
		return new PersonaId(value);
	}

	public static PersonaId defaultPersona() {
		return DEFAULT;
	}

	public static PersonaId ofNullable(String value) {
		return (value == null || value.isBlank()) ? DEFAULT : new PersonaId(value);
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/dialogue/model/PersonaId.kt
package com.miyou.app.domain.dialogue.model

data class PersonaId(
    val value: String,
) {
    init {
        require(value.isNotBlank()) { "personaId cannot be null or blank" }
        require(value.length <= 64) { "personaId too long" }
        require(value.matches(Regex("^[a-zA-Z0-9_-]+$"))) {
            "personaId must contain only alphanumeric characters, hyphens, and underscores"
        }
    }

    companion object {
        val DEFAULT: PersonaId = PersonaId("default")

        @JvmStatic
        fun of(value: String): PersonaId = PersonaId(value)

        @JvmStatic
        fun defaultPersona(): PersonaId = DEFAULT

        @JvmStatic
        fun ofNullable(value: String?): PersonaId = if (value.isNullOrBlank()) DEFAULT else PersonaId(value)
    }

    fun value(): String = value
}
